### PR TITLE
Add impl for torchvision.nms and support multiple batch dim for convolution.

### PR DESCRIPTION
**Pull request by qihqi**
This is partial progress to support MaskRCNN (#8116).

Fixes: #8278

### CODE REVIEW

File: experimental/torch_xla2/torch_xla2/ops/jaten.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ def _aten_add(x, y, *, alpha=1):
 
   assert x.dtype == y.dtype, (x.dtype, y.dtype)
   """
+  if isinstance(x, torch.Tensor) or isinstance(y, torch.Tensor):
+    import pdb; pdb.set_trace()
```
Comment (ManfeiBai) [inline experimental/torch_xla2/torch_xla2/ops/jaten.py:113]: do we want to keep this here?

File: experimental/torch_xla2/torch_xla2/ops/jtorch.py [hunk 2]
```diff
@@ -98,13 +99,11 @@ def get_params(*a):
   return jnp.einsum(equation, *filtered_operands)
 
 
-def _sdpa_reference(
-   query, key, value, attn_mask=None,
-   dropout_p=0.0, is_causal=False, scale=None) -> torch.Tensor:
+def _sdpa_reference(query, key, value, attn_mask=None, dropout_p=0.0,
+        is_causal=False, scale=None, enable_gqa=False) -> torch.Tensor:
```
Comment (ManfeiBai) [inline experimental/torch_xla2/torch_xla2/ops/jtorch.py:103]: thanks, do we have short description for `enable_gpa`?

File: experimental/torch_xla2/torch_xla2/tensor.py [hunk 3]
```diff
@@ -166,14 +166,20 @@ def tolist(self):
 def debug_accuracy(func, args, kwargs, current_output):
   args_torch, kwargs_torch, out_torch = torch_pytree.tree_map_only(
       torch.Tensor, lambda x: j2t(x._elem), (args, kwargs, current_output))
+  
+  if func == torch.ops.aten._native_batch_norm_legit_no_training:
+    import pdb; pdb.set_trace()
```
Comment (ManfeiBai) [inline experimental/torch_xla2/torch_xla2/tensor.py:171]: thanks, do we want to keep this code here?

File: experimental/torch_xla2/torch_xla2/ops/jtorch.py (hunk 2 above)
Comment (qihqi) [reply experimental/torch_xla2/torch_xla2/ops/jtorch.py:103]: yes: https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html
this is newly added in torch 2.5
